File: DSA/pranavdixit_590014418/Day_27_Question_2.c
def reverseDLL(head):
    current = head
    prev_node = None

    while current:
        current.prev, current.next = current.next, current.prev
        prev_node = current
        current = current.prev

    return prev_node

# Sample Test
n1 = Node(3)
n2 = Node(4)
n3 = Node(5)
n1.next = n2
n2.prev = n1
n2.next = n3
n3.prev = n2

head = reverseDLL(n1)
printList(head)

File: DSA/PranyaGupta_500126032/Day27Question2.py
class node:
    def __init__(self, data):
        self.next = None
        self.prev = None
        self.data = data
class LinkedList:
    def __init__(self):
        self.head = None
    def add(self, data):
        new_node = node(data)
        if not self.head:
            self.head = new_node
            return
        current = self.head
        while current.next:
            current = current.next
        current.next = new_node
        new_node.prev = current
    def printlist(self, head):
        current = head
        while current:
            print(current.data, end=" ")
            current = current.next   
    def reverse(self, head):
        
        current = self.head
        temp = None
        while current:
            temp = current.prev
            current.prev = current.next
            current.next = temp
            current = current.prev
        self.head = temp.prev    
        self.printlist(self.head)   
   


def main():
    user_input = input("Enter elements of the linked list (with space in between): ")
    arr = list(map(int, user_input.strip().split()))
    l = LinkedList()
    for num in arr:
        l.add(num)
    l.reverse(l.head)
if __name__ == "__main__":
    main()